### PR TITLE
Improve non-proxy error

### PR DESCRIPTION
**Pull request by Aslemammad**
@dai-shi, I think we should write tests for errors, What's your idea?

### CODE REVIEW

**Comment by dai-shi** [comment]
> I think we should write tests for errors, What's your idea?

It's definitely desirable.

**Comment by Aslemammad** [comment]
> > I think we should write tests for errors, What's your idea?
> 
> It's definitely desirable.

I'll work on it in an another PR.

**Comment by Aslemammad** [comment]
@dai-shi How is it now?

File: src/vanilla.ts [hunk 1]
```diff
@@ -4,6 +4,9 @@ const VERSION = Symbol()
 const LISTENERS = Symbol()
 const SNAPSHOT = Symbol()
 
+const isProduction =
+  typeof process === 'object' && process.env.NODE_ENV === 'production'
```
Comment (dai-shi) [inline src/vanilla.ts:8]:
```suggestion
  typeof process !== 'object' || process.env.NODE_ENV === 'production'
```
You failed to invert the logic. 😄

**Review by dai-shi — CHANGES_REQUESTED** [review]
As far as I confirm with https://xem.github.io/terser-online/,
`isProduction` fails to minify the production code.
Please make it inline too.

File: src/vanilla.ts (hunk 1 above)
Comment (Aslemammad) [reply src/vanilla.ts:8]: Yea, I forgot. thanks.

**Comment by Aslemammad** [comment]
@dai-shi I made isProduction inline now.

**Review by dai-shi — APPROVED** [review]
Thanks for adjustment!

**Comment by Aslemammad** [comment]
@dai-shi Thanks.

**Comment by dai-shi** [comment]
What would happen if one does `useProxy({})`?
We'd want to invoke `getVersion()` at the beginning. (and we need another `if`. 🤷‍♂️ )

**Comment by Aslemammad** [comment]
jest fail tests: Error: Please use proxy object Jest in useProxy

**Comment by dai-shi** [comment]
That's good. Let's merge this in.
Thanks for your contribution!

**Comment by Aslemammad** [comment]
Thanks for the merging.

**Pull request merged**